core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/DetailedSampleBuilder.java

```diff
@@ -824,17 +827,14 @@ public void setNonStandardAlias(boolean nonStandardAlias) {
     this.nonStandardAlias = nonStandardAlias;
   }
 
-  public DetailedSample build() {
+  public DetailedSample build(InferStockFromAliquot inferencer) {
```

Comment:
This is not always needed. Please keep build method parameterless. There can be a setter for this and an exception thrown if it's needed and not set